frontend/src/pages/modelServing/screens/projects/NIMServiceModal/DeployNIMServiceModal.tsx

```diff
@@ -123,16 +126,43 @@ const DeployNIMServiceModal: React.FC<DeployNIMServiceModalProps> = ({
   const [error, setError] = React.useState<Error | undefined>();
   const [alertVisible, setAlertVisible] = React.useState(true);
   const [pvcSize, setPvcSize] = React.useState<string>('30Gi');
+  const [existingPvcSize, setExistingPvcSize] = React.useState<string | null>(null);
+  const [existingPVC, setExistingPVC] = React.useState<PersistentVolumeClaimKind | undefined>();
+  const [createData, setCreateData, resetData] = useCreateStorageObjectForNotebook(existingPVC);
 
   React.useEffect(() => {
     if (currentProjectName) {
       setCreateDataInferenceService('project', currentProjectName);
     }
   }, [currentProjectName, setCreateDataInferenceService]);
 
+  React.useEffect(() => {
+    const fetchPVCSize = async () => {
+      if (editInfo && editInfo.inferenceServiceEditInfo?.metadata.namespace) {
+        const pvcData = await getPVC(
+          editInfo.inferenceServiceEditInfo.metadata.namespace,
+          editInfo,
+        );
+        if (pvcData) {
+          setExistingPVC(pvcData);
+          const size = pvcData.spec.resources.requests.storage;
+          if (size) {
+            setExistingPvcSize(size);
+            setPvcSize(size);
+          }
+        }
+      }
+    };
+
+    fetchPVCSize();
+  }, [namespace, editInfo]);
+
+  React.useEffect(() => {
+    setCreateData('size', pvcSize);
+  }, [pvcSize, setCreateData]);
+
   // Serving Runtime Validation
-  const isDisabledServingRuntime =
-    namespace === '' || actionInProgress || createDataServingRuntime.imageName === undefined;
+  const isDisabledServingRuntime = namespace === '' || actionInProgress;
```

Comment:
Was this a mistake from copying from KServe? Since NIM runtime always has a image name?